datus/storage/metric/store.py

```
@@ -174,6 +174,101 @@ def search_all(
+                        result["yaml_updated"] = True
+                        result["message"] = f"Deleted metric '{name}' from lancedb and yaml file"
+                        logger.info(f"Updated yaml file: {yaml_path}")
+                    else:
+                        # File is empty after removing the metric, delete the file
+                        os.remove(yaml_path)
+                        result["yaml_updated"] = True
+                        result["yaml_deleted"] = True
+                        result["message"] = f"Deleted metric '{name}' from lancedb and removed empty yaml file"
+                        logger.info(f"Deleted empty yaml file: {yaml_path}")
+
+            except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)